plasma_framework/test/src/vaults/Erc20Vault.test.js

```diff
@@ -0,0 +1,139 @@
+            );
+        });
+
+        it('should not store an ethereum deposit that sends funds', async () => {
+            const deposit = Testlang.deposit(DEPOSIT_VALUE, alice);
+
+            await expectRevert.unspecified(
+                this.erc20Vault.deposit(deposit, { from: alice, value: DEPOSIT_VALUE }),
+            );
+        });
+
+        it('should not store an ethereum deposit that does not send funds', async () => {
+            const deposit = Testlang.deposit(DEPOSIT_VALUE, alice);
+
+            await expectRevert(
+                this.erc20Vault.deposit(deposit, { from: alice }),
+                'Invalid output currency (ETH)',
+            );
+        });
+
+        it('should not accept transaction that does not match expected transaction type', async () => {
+            const output = new PaymentTransactionOutput(DEPOSIT_VALUE, alice, this.erc20.address);
+            const WRONG_TX_TYPE = 123;
+            const deposit = new PaymentTransaction(WRONG_TX_TYPE, [0], [output]);
+
+            await expectRevert(
+                this.erc20Vault.deposit(deposit.rlpEncoded(), { from: alice }),
+                'Invalid transaction type.',
+            );
+        });
+
+        it('should not accept transaction that does not conform to deposit input format', async () => {
+            const invalidInput = Buffer.alloc(32, 1);
+            const output = new PaymentTransactionOutput(DEPOSIT_VALUE, alice, this.erc20.address);
+            const deposit = new PaymentTransaction(1, [invalidInput], [output]);
+
+            await expectRevert(
+                this.erc20Vault.deposit(deposit.rlpEncoded(), { from: alice }),
+                'Deposit input must be bytes32 of 0',
+            );
+        });
+
+        it('should not accept transaction with more than one output', async () => {
+            const output = new PaymentTransactionOutput(DEPOSIT_VALUE, alice, this.erc20.address);
+            const deposit = new PaymentTransaction(1, [0], [output, output]);
+
+            await expectRevert(
+                this.erc20Vault.deposit(deposit.rlpEncoded(), { from: alice }),
+                'Must have only one output',
+            );
+        });
+    });
+
+    describe('deposit from BadERC20', () => {
+        // A 'BadERC20' token is one that uses an old version of the ERC20 standard,
+        // as described here https://medium.com/coinmonks/missing-return-value-bug-at-least-130-tokens-affected-d67bf08521ca
+        // Erc20Vault should support both versions.
+        before('setup', async () => {
+            this.badErc20 = await BadERC20.new(INITIAL_SUPPLY);
+            await this.badErc20.transfer(alice, DEPOSIT_VALUE, { from: accounts[0] });
+        });
+
+        it('should store erc20 deposit', async () => {
+            await this.badErc20.approve(this.erc20Vault.address, DEPOSIT_VALUE, { from: alice });
+            let nextDepositBlock = parseInt(await this.blockController.nextDepositBlock(), 10);
```

Comment:
Let's compare expected and actual deposit blocks as in `should store erc20 deposit` test.